src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -129,7 +129,15 @@ public OAuthHandler(HttpClient backchannel)
                 properties.StoreTokens(authTokens);
             }
 
-            return AuthenticateResult.Success(await CreateTicketAsync(identity, properties, tokens));
+            try
+            {
+                var ticket = await CreateTicketAsync(identity, properties, tokens);
+                return AuthenticateResult.Success(ticket);
+            }
+            catch(Exception ex)
```

Comment:
Actually I think this is a good spot. The other parts of this code can handle their own errors and return AuthenticateResult.Fail, but CreateTicketAsync can't.